src/components/authorization/Authorization.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { useState, useEffect } from "react";
+import i18n from "../../locales";
+import { Typography } from "@material-ui/core";
+
+interface AuthorizationProps {
+    authorize: () => Promise<boolean>;
+}
+
+const Authorization: React.FC<AuthorizationProps> = props => {
+    const [isAuthorize, setIsAuthorize] = useState<boolean>(true);
+
+    useEffect(() => {
+        async function executeIsAuthorize() {
+            const authorized = await props.authorize();
+
+            setIsAuthorize(authorized);
+        }
+
+        executeIsAuthorize();
+    }, [props]);
```

Comment:
1. Adding ``props`` to the dependency array might lead to duplicated side effects. Instead ``props.authorize`` would be the correct dependency (or just ``authorize`` if we destructure the props object).

2. A cleaner way to handle async operations in useEffect is to use ``then``: 

```
useEffect(() => {
    authorize().then(setIsAuthorize);
}, [authorize]);
```

Reply:
Done!